cpp/src/io/json/json_column.cu

```diff
@@ -686,7 +744,11 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> device_json_co
       auto [child_column, names] =
         json_col.child_columns.empty()
           ? std::pair<std::unique_ptr<column>,
-                      std::vector<column_name_info>>{std::make_unique<column>(), {}}
+                      std::vector<column_name_info>>{std::make_unique<column>(
+                                                       data_type{type_id::INT8},
+                                                       0,
+                                                       rmm::device_buffer{0, stream, mr}),
```

Comment:
Oh why? The lists column has empty child, right?

Reply:
I don't see this change making any difference, except that the new child type is now `INT8` instead of `EMPTY`. This may avoid exception thrown by `type_dispatcher` in some code paths. But IMO I think keeping the return type of `EMPTY` column is more appropriate.

Reply:
Type dispatcher fails when the child column is empty. That's why it's defaulted to INT8.

Reply:
The `EMPTY` type is likely to be removed: https://github.com/rapidsai/cudf/issues/12477

Reply:
> Type dispatcher fails when the child column is empty. That's why it's defaulted to INT8.

Can you also add a comment quoting this please?

Reply:
@ttnghia as discussed offline, gather throws error when children is type_id::EMPTY, during `purge_nonempty_nulls`.
Also, For expecting future gather operations on the resulting list column, we are choosing to return INT8 of size 0 here.


Also, `purge_nonempty_nulls` could be improved to skip the gather if it does not have non-empty nulls.